src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala

```diff
@@ -261,70 +284,96 @@ object SbtBundle extends AutoPlugin {
   }
 
-  private def getConfig: Def.Initialize[Task[String]] = Def.task {
-    val checkInitialDelayInSeconds =
-      if (checkInitialDelay.value.toMillis % 1000 > 0)
-        checkInitialDelay.value.toSeconds + 1 // always round up
-      else
-        checkInitialDelay.value.toSeconds
-    val checkComponents = if (checks.value.nonEmpty)
-      checks.value.map(uri => s""""$uri"""").mkString(
-        s""",
-           |  "${(normalizedName in Bundle).value}-status" = {
-           |    description      = "Status check for the bundle component"
-           |    file-system-type = "universal"
-           |    start-command    = ["check", "--initial-delay", "$checkInitialDelayInSeconds", """.stripMargin,
-        ", ",
-        s"""]
-           |    endpoints        = {}
-           |  }""".stripMargin)
-    else
-      ""
-
-    s"""|version              = "1.1.0"
-        |name                 = "${(normalizedName in Bundle).value}"
-        |compatibilityVersion = "${compatibilityVersion.value}"
-        |system               = "${system.value}"
-        |systemVersion        = "${systemVersion.value}"
-        |nrOfCpus             = ${nrOfCpus.value}
-        |memory               = ${memory.value.underlying}
-        |diskSpace            = ${diskSpace.value.underlying}
-        |roles                = ${formatSeq(roles.value)}
-        |components = {
-        |  "${(normalizedName in Bundle).value}" = {
-        |    description      = "${projectInfo.value.description}"
-        |    file-system-type = "${bundleType.value}"
-        |    start-command    = ${formatSeq(startCommand.value)}
-        |    endpoints        = ${formatEndpoints(endpoints.value)}
-        |  }$checkComponents
-        |}
-        |""".stripMargin
+  private def getConfig(config: Configuration, filter: ScopeFilter): Def.Initialize[Task[String]] = Def.task {
+    val checkComponents = checks.?.all(filter).value.flatMap {
+      case Some(checksValue) if checksValue.nonEmpty =>
+        val checkInitialDelayValue = (checkInitialDelay in config).value
+        val checkInitialDelayInSeconds =
+          if (checkInitialDelayValue.toMillis % 1000 > 0)
+            checkInitialDelayValue.toSeconds + 1 // always round up
+          else
+            checkInitialDelayValue.toSeconds
+        checksValue.map(uri => s""""$uri"""").mkString(
+          s"""components."${(normalizedName in config).value}-status" = {
+             |  description      = "Status check for the bundle component"
+             |  file-system-type = "universal"
+             |  start-command    = ["check", "--initial-delay", "$checkInitialDelayInSeconds", """.stripMargin,
+          ", ",
+          s"""]
+             |  endpoints        = {}
+             |}""".stripMargin)
+      case _ =>
+        Seq.empty[String]
+    }
+
+    val declarations =
+      normalizedName.?.all(filter).value.collect {
+        case Some(value) => s"""name = "$value""""
+      } ++
```

Comment:
Probably not possible to change the way the key evaluates.
A workaround might be to define your own `String` key instead of using `normalizedName`, so it will always pick up the one scoped for the right configuration.